src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs

```diff
@@ -437,6 +437,11 @@ public virtual string CustomMethod
         /// </summary>
         private string _originalFilePath;
 
+        /// <summary>
+        /// True if the web session was created specifically by this request
+        /// </summary>
+        private bool _noWebSession;
```

Comment:
This is used only once for conditional disposing HttpClientHandler.
I suggest to keep HttpClient in WebSession too since they exist only in couple.
Then _noWebSession can be removed.